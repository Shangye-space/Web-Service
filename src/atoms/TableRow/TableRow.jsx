import React from 'react';

export default function TableRow({
    id,
    img,
    name,
    description,
    price,
    in_sale,
    quantity,
    discount,
}) {
    return (
        <tr>
            <th scope="row">{id}</th>
            <td>
                <img style={{ height: '60px', width: '60px' }} src={img} />
            </td>
            <td>{name}</td>
            <td>{description}</td>
            <td>{price}</td>
            <td>{in_sale.toString()}</td>
            <td>{quantity}</td>
            <td>{discount}</td>
            <td></td>
        </tr>
    );
}
