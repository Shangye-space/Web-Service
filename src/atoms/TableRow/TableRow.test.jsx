import React from 'react';
import ReactDOM from 'react-dom';
import TableRow from './TableRow.jsx';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<TableRow in_sale={true} />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('snapshot renders', () => {
    const component = renderer.create(<TableRow in_sale={true} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
