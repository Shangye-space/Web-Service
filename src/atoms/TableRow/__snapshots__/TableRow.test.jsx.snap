// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders 1`] = `
<tr>
  <th
    scope="row"
  />
  <td>
    <img
      style={
        Object {
          "height": "60px",
          "width": "60px",
        }
      }
    />
  </td>
  <td />
  <td />
  <td />
  <td>
    true
  </td>
  <td />
  <td />
  <td />
</tr>
`;
